### DISCUSSION TITLE
After adding socketio.RedisManager, the connection is not successful.

### DISCUSSION THREAD

**Discussion by JuneDan** [miguelgrinberg/python-socketio#1199] [Q&A] [answered]
I am a Python beginner and I am experiencing some issues while using this library. Can you please help me? Thank you very much.

python socketIO server ：

```
import eventlet
import socketio

mgr = socketio.RedisManager()
sio = socketio.Server(client_manager=mgr, logger=True, engineio_logger=True)
app = socketio.WSGIApp(sio)

@sio.event
def connect(sid, environ):
    print('New connection',sid)

@sio.event
def my_message(sid: str, question: str):
    sio.emit('on_serve_msg',‘ok’)
    

@sio.event
def disconnect(sid):
    print('disconnect ', sid)

if __name__ == '__main__':
    eventlet.wsgi.server(eventlet.listen(('', 5000)), app)
```
python socketIO client ：
```
import socketio
sio = socketio.Client(logger=True, engineio_logger=True)

@sio.event
def connect():
    while True:
        string = input('>>')
        my_message(string)

@sio.event
def my_message(data):
    sio.emit('my_message', data)

@sio.event
def disconnect():
    print('disconnected from server')

sio.connect('ws://localhost:5000')
sio.wait()

```
error（client）：
```
Attempting polling connection to http://localhost:5000/socket.io/?transport=polling&EIO=4
HTTP GET request to http://localhost:5000/socket.io/?transport=polling&EIO=4&t=1686195897.0514188 failed with error HTTPConnectionPool(host='localhost', port=5000): Read timed out. (read timeout=5).
Traceback (most recent call last):
  File "C:\Users\10351\Desktop\superGPT\gpt_langchain_py\testConnect\eventletSocketIOc.py", line 30, in <module>
    sio.connect('ws://localhost:5000')
  File "C:\Program Files\python\Lib\site-packages\socketio\client.py", line 338, in connect
    raise exceptions.ConnectionError(exc.args[0]) from None
socketio.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=5000): Read timed out. (read timeout=5)
```

**Comment by miguelgrinberg** [comment]
Did you look at your server logs for clues?

**Reply by JuneDan** [reply]
Thank you very much for your reply. I have checked the server logs and found no abnormalities. (Or where should I look for more detailed logs?) The server output is as follows:
```
Server initialized for eventlet.
(20948) wsgi starting up on http://0.0.0.0:5000
(20948) accepted ('127.0.0.1', 4385)
EorcL_bRsACHCsN9AAAA: Sending packet OPEN data {'sid': 'EorcL_bRsACHCsN9AAAA', 'upgrades': ['websocket'], 'pingTimeout': 20000, 'pingInterval': 25000}
```

**Reply by miguelgrinberg** [reply]
Are you running Redis on your server?

**Reply by JuneDan** [reply]
Yes, Redis is running. The default port is 6379. Because I connect to it using Node.js and it runs normally.

**Reply by JuneDan** [reply]
May I ask if I need to provide more detailed code?

**Reply by JuneDan** [reply]
I forgot to mention that Redis is running in Docker, and the mapped port is also 6379.

**Reply by miguelgrinberg** [reply]
I would suggest adding logs for your web server (passing `log_output=True` I think, assuming this is eventlet's WSGI server), so that there is more information on your server logs. The logs that you shared indicate a successful connection, not a connection error.

**Reply by JuneDan** [reply, marked as answer]
Thank you very much for the suggestion, but the server still did not generate any error logs. However, I have solved this problem. I added eventlet.monkey_patch() in the SocketIO server , and the connection was successful. It's amazing, and I'm not sure why. If you know the reason, please let me know. Thank you very much.

**Reply by miguelgrinberg** [reply]
This is correct. Monkey patching is required when you use eventlet and redis combined. This is mentioned in the documentation, and there is also a runtime error that occurs when monkey patching is not detected:

```
RuntimeError: Redis requires a monkey patched socket library to work with eventlet
```

I have tested the above example code you provided and I do get the error. Not sure why you did not see this on your side.
